src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,28 @@
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        lock.readLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+            lock.readLock().unlock();
```

Comment:
You don't need to unlock the read lock after performing a write operation. Remove the line `lock.readLock().unlock();`. 😊